.github/workflows/deploy.yml

```
@@ -0,0 +1,117 @@
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  # Backend deployment to Google Cloud Functions
+  deploy-backend:
+    runs-on: ubuntu-latest
+    if: contains(github.event.head_commit.modified, 'backend/') || github.event_name == 'workflow_dispatch'
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)